src/translation-service.js

```diff
@@ -9,62 +9,127 @@
  * @module utils/translation-service
  */
 
+import CKEditorError from './ckeditorerror';
+
 /* istanbul ignore else */
 if ( !window.CKEDITOR_TRANSLATIONS ) {
 	window.CKEDITOR_TRANSLATIONS = {};
 }
 
 /**
- * Adds translations to existing ones.
- * These translations will later be available for the {@link module:utils/translation-service~translate `translate()`} function.
+ * Adds translations to existing ones or overrides the existing translations. These translations will later
+ * be available for the {@link module:utils/locale~Locale#t `t()`} function.
+ *
+ * The `translations` is an object which consists of a `messageId: translation` pairs. Note that the message id can be
+ * either constructed either from the message string or from the message string and the message context in the following form:
+ * `<messageString>_<messageContext>` (this happens rarely and mostly for short messages or messages with placeholders).
+ * Since the editor displays only the message string, the message context can be found either in the source code or in the
+ * built translations for another language.
  *
  *		add( 'pl', {
- *			'OK': 'OK',
- *			'Cancel [context: reject]': 'Anuluj'
+ *			'Cancel': 'Anuluj',
+ *			'image_Insert image': 'obraz', // Note that the `Insert image` comes from the message context.
  *		} );
  *
+ * If the message is supposed to support various plural forms, make sure to provide an array with the single form and all plural forms:
+ *
+ *		add( 'pl', {
+ *	 		'Add space': [ 'Dodaj spację', 'Dodaj %0 spacje', 'Dodaj %0 spacji' ]
```

Comment:
What is the equivalent of adding it using our `t()` function?

In the docs I've seen only like:

```plaintext
translate( 'en', { string: 'Add a space', plural: 'Add %0 spaces' }, 3 );
```

Which seems to be limited to one plural form.

Reply:
The plural form in a translation message specifies that the message supports plural version and specifies the default plural form for missing translations. It doesn't specifies the number of plural forms.

So the `translate( 'pl', { string: 'Add a space', plural: 'Add %0 spaces' }, 3 );` should return `Dodaj 3 spacje` in your example.

Reply:
OK, so to ensure I get this right: `t()` can pass only one plural form (because English supports only one plural form) thus multiple plural forms will be _always_ added using the [`translation-service#add()`](https://ckeditor.com/docs/ckeditor5/18.0.0/api/module_utils_translation-service.html#static-function-add) method, right?

If that's the case, this conversation is resolved.

Reply:
Right!